### ISSUE TITLE
Add version to header comment.

### ISSUE THREAD

**Issue by jenstroeger** [grpc/grpc#31765]
### Is your feature request related to a problem? Please describe.

Frequent version changes of the tools require generated code to keep in step with tools changes. That can be difficult to maintain.

### Describe the solution you'd like

Generated code should contain the version number of the `protoc` it was built with, e.g. for generated Python code:
```python
# Generated by the protocol buffer compiler: libprotoc 3.19.5.  DO NOT EDIT! 
```

### Describe alternatives you've considered

I currently wrap the `protoc` invocation in a shell script and use `sed` to inject that version number.

### Additional context

n/a

**Comment by gnossen** [comment]
To be clear, this is only in our power for `_pb2_grpc.py` files, not `_pb2.py` files. But I think this sounds like a great idea. We'd likely want to include `grpcio-tools` version for our end of things.

You should file a separate feature request on the protobuf repo for the addition to `_pb2.py` files.

**Comment by jenstroeger** [comment]
> To be clear, this is only in our power for `_pb2_grpc.py` files, not `_pb2.py` files.

Oh interesting, I assumed both were generated by `protoc`?

> But I think this sounds like a great idea. We'd likely want to include `grpcio-tools` version for our end of things.

Nice 👍🏼 

> You should file a separate feature request on the protobuf repo for the addition to `_pb2.py` files.

That’s the [protocolbuffers/protobuf](https://github.com/protocolbuffers/protobuf) repo?

**Comment by gnossen** [comment]
> Oh interesting, I assumed both were generated by protoc?

Not quite. `protoc` itself generates the `_pb2.py` and `_pb2.pyi` files. The gRPC Python `protoc` plugin (which is bundled together with `protoc` by the `grpcio-tools` package) generates the `_pb2_grpc.py` files.

> That’s the [protocolbuffers/protobuf](https://github.com/protocolbuffers/protobuf) repo?

Yes.

**Comment by alielbashir** [comment]
Is this issue meant for generated files in all languages, or only Python?

**Comment by jenstroeger** [comment]
I’m primarily interested in Python only but I could see how this is useful across all languages…

**Comment by alielbashir** [comment]
@jenstroeger I did a Python implementation for this in protoc in https://github.com/protocolbuffers/protobuf/pull/11408

Depending on what the gRPC and Protobuf team request, I can also work on this feature for all languages on both protoc and gRPC

**Comment by jenstroeger** [comment]
Great, thank you for that!
